spec-dtslint/observables/defer-spec.ts

```diff
@@ -28,6 +28,10 @@ it('should error if function returns undefined', () => {
   const a = defer(() => undefined); // $ExpectError
 });
 
+it('should error if function returns never', () => {
+  const a = defer(() => { throw new Error(); }); // $ExpectError
+});
```

Comment:
Does this work?

```ts
defer(() => {
   if (Math.random() > 0.5) {
      throw new Error();
   }
   return of(1, 2, 3);
});
```

I mean, I assume it does, but It would be nice to have a test there.

Reply:
Added